src/features/posts/usePosts.ts

```diff
@@ -1,15 +1,27 @@
+import { useEffect } from "react";
 import { useAppDispatch, useAppSelector } from "../../redux/hooks";
-import { fetchPosts, likePost, dislikePost, favoritePost } from "./postsSlice";
+import { fetchOnePost, fetchPosts, likePost, dislikePost, favoritePost } from "./postsSlice";
 
 export const usePosts = () => {
     const posts = useAppSelector( state => state.posts.content)
     
+    const onePost = useAppSelector( state => state.posts.post)
+
     const dispatch = useAppDispatch()
     
-    if(!posts){
-        dispatch(fetchPosts())
-    }
+    useEffect(() => {
+        {!posts 
+        && 
+        dispatch(fetchPosts())}
+    })
```

Comment:
Обновлять посты тебе нужно на данном этапе только при после первого монтирования компонента, поэтому имеем []. И лучше не сокращать, если это влияет на читаемость кода
```suggestion
    useEffect(() => {
        if (!posts) {
          dispatch(fetchPosts())}
        }
    }, [])
```